public/app/features/trails/AutomaticMetricQueries/units.ts

```diff
@@ -9,9 +9,15 @@ export function getUnitFromMetric(metric: string) {
     return suffix;
   } else if (UNIT_LIST.includes(secondToLastSuffix)) {
     return secondToLastSuffix;
-  } else {
-    return null;
   }
+  return undefined;
+}
+
+// Get unit from recording rule (e.g. "workload:memory_working_set_bytes:sum" -> "bytes")
+export function getUnitFromRecordingRule(recordingRule: string) {
+  const lowerCaseRule = recordingRule.toLowerCase();
+
+  return Object.keys(UNIT_MAP).find((key) => lowerCaseRule.includes(key));
```

Comment:
This will detect `milliseconds` as `seconds`.

We can consider the recording rule format of `level:metric:operations` as outlined here: https://prometheus.io/docs/practices/rules/#recording-rules
- Break it up by `:`, and then only consider the inner `metric`...

But even that won't be perfect since recording rules map to expressions which might consist of multiple metrics and operators between them.

I am reconsidering whether or not this is a feasible task without requiring metric metadata.